Sourcery/Generating/Generator.swift

```diff
@@ -101,10 +93,39 @@ extension Type {
 }
 
 enum Generator {
+    private static func updateTypeRelationship(for type: Type, typesByName: [String: Type], processed: inout [String: Bool]) {
+        type.based.keys.forEach { name in
+            guard let baseType = typesByName[name] else { return }
```

Comment:
I would say to make it work it should be`guard let baseType = typesByName[name] else { type.based[name] = name; return }`